src/js/get-attribute.js

```diff
@@ -1,15 +1,20 @@
 import toProp from './to-prop';
+import camelize from './camelize';
 
 /**
  * Get an HTML attribute's value by it's name.
  * **Note:** The attribute's value get parsed as JSON if it's valid JSON.
  *
  * @param {Element|Attribute} node - The HTML element to get an attribute from.
  * @param {String} [name] - The attribute's name.
+ * @param {String} [type] - The attribute's type.
  * @returns {*} - Returns the value of the attribute - in case of valid JSON the result of `JSON.parse` else plain text.
  */
-function getAttribute(node, name) {
-  if (typeof node.hasAttribute === 'function' && !node.hasAttribute(name)) {
+function getAttribute(node, name, type) {
+  let derivedType = type;
+  const isElementNode = node.nodeType === 1;
+
+  if (isElementNode && !node.hasAttribute(name)) {
     return false;
```

Comment:
@LucaMele 
I just saw this, we return `false` if an attribute does not exists, though this should only be the case for `PropTypes.bool` 💣 
All others should be `undefined` 🤔